ipywidgets/widgets/widget.py

```diff
@@ -77,8 +80,8 @@ def register_callback(self, callback, remove=False):
         
         # (Un)Register the callback.
         if remove and callback in self.callbacks:
-            self.callbacks.remove(callback)
-        elif not remove and callback not in self.callbacks:
```

Comment:
I think, like we talked about in person, I'd rather keep this bit- which I think you said you agreed.